clang/lib/AST/Type.cpp

```diff
@@ -4855,6 +4857,17 @@ LinkageInfo LinkageComputer::computeTypeLinkageInfo(const Type *T) {
     return computeTypeLinkageInfo(cast<HLSLAttributedResourceType>(T)
                                       ->getContainedType()
                                       ->getCanonicalTypeInternal());
+  case Type::HLSLInlineSpirv:
+    return LinkageInfo::external();
+    {
```

Comment:
What's going on here? We always return but we've got some dead code following this. (I suspect we're missing some test coverage for non-external linkages.

Reply:
I don't know enough about HLSL to know what the right answer is here, so hopefully @llvm-beanz can step in for that.

Reply:
https://github.com/llvm/llvm-project/pull/146365